requests/models.py

```diff
@@ -813,7 +813,10 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             #
             # If we're using `splitlines()`, we only do this if the chunk
             # ended midway through a line.
-            incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            if isinstance(chunk[-1], basestring):  # Decoded string (decode_unicode=True or Py2)
+                incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            else:  # Bytestring (decode_unicode=False and Py3)
+                incomplete_line = lines[-1] and lines[-1].endswith(chr(chunk[-1]).encode('utf-8'))
```

Comment:
I'm confused: I checked in a normal usage (outside of the tests) and could reproduce the issue:
with Python2 I get `str`, and with Python3 I get `bytes` (and if I use `iter_lines(decode_unicode=True)`
I get `str` (Py3) or `unicode` (Py2)

The previous code wasn't accepting bytes (and was working only with Python 2)

This failed on Python3 before my last fix:
```
r = requests.get("https://github.com", stream=True)
list(r.iter_lines())
```
TypeError: endswith first arg must be bytes or a tuple of bytes, not int

That's why I don't think the tests were in error (a test that I didn't write was failing, `test_response_iter_lines`)